contracts/V3Migrator.sol

```diff
@@ -13,19 +13,27 @@ import './libraries/TransferHelper.sol';
 
 import './libraries/LiquidityAmounts.sol';
 import './interfaces/IV3Migrator.sol';
+import './base/PeripheryImmutableState.sol';
 import './base/Multicall.sol';
 import './base/SelfPermit.sol';
+import './interfaces/external/IWETH9.sol';
 
 /// @title Uniswap V3 Migrator
-contract V3Migrator is IV3Migrator, Multicall, SelfPermit {
-    address public immutable factory;
+contract V3Migrator is IV3Migrator, PeripheryImmutableState, Multicall, SelfPermit {
     address public immutable nonfungiblePositionManager;
 
-    constructor(address _factory, address _nonfungiblePositionManager) {
-        factory = _factory;
+    constructor(
+        address _factory,
+        address _WETH9,
+        address _nonfungiblePositionManager
+    ) PeripheryImmutableState(_factory, _WETH9) {
         nonfungiblePositionManager = _nonfungiblePositionManager;
     }
 
+    receive() external payable {
```

Comment:
should this be inherited?

Reply:
i don't want to add the sweep methods into here though, as we have an explicit param for handling eth and i don't want to confuse people